client/core/status.go

```diff
@@ -0,0 +1,382 @@
+	{order.NewlyMatched, order.MakerSwapCast, resolveMissedMakerAudit},
+	{order.MakerSwapCast, order.NewlyMatched, resolveServerMissedMakerInit},
+	{order.MakerSwapCast, order.TakerSwapCast, resolveMissedTakerAudit},
+	{order.TakerSwapCast, order.MakerSwapCast, resolveServerMissedTakerInit},
+	{order.TakerSwapCast, order.MakerRedeemed, resolveMissedMakerRedemption},
+	{order.MakerRedeemed, order.TakerSwapCast, resolveServerMissedMakerRedeem},
+	{order.MakerRedeemed, order.MatchComplete, resolveMissedTakerRedemption},
+	{order.MatchComplete, order.MakerRedeemed, resolveServerMissedTakerRedeem},
+}
+
+// conflictResolver is a getter for a matchConflictResolver for the specified
+// MatchStatus combination. If there is no resolver for this combination, a
+// nil resolver will be returned.
+func conflictResolver(ours, servers order.MatchStatus) matchConflictResolver {
+	for _, r := range conflictResolvers {
+		if r.ours == ours && r.servers == servers {
+			return r.resolver
+		}
+	}
+	return nil
+}
+
+// resolveConflictWithServerData compares the match status with the server's
+// match_status data. The trackedTrade.mtx is locked for the duration of
+// resolution. If the conflict cannot be resolved, the match will be
+// self-revoked.
+func (c *Core) resolveConflictWithServerData(dc *dexConnection, trade *trackedTrade, match *matchTracker, srvData *msgjson.MatchStatusResult) {
+	srvStatus := order.MatchStatus(srvData.Status)
+
+	if srvStatus == match.MetaData.Status || match.MetaData.Proof.IsRevoked() {
+		// On startup, there's no chance for a tick between the connect request
+		// and the match_status request, so this would be unlikely. But if not
+		// during startup, and a tick has snuck in and resolved our status
+		// conflict already (either by refunding or via resendPendingRequests),
+		// that's OK.
+		return
+	}
+
+	logID := statusResolutionID(dc, trade, match)
+
+	resolver := conflictResolver(match.MetaData.Status, srvStatus)
+	if resolver != nil {
+		resolver(c, dc, trade, match, srvData)
+	} else {
+		// We don't know how to handle this. Set the failErr, and self-revoke
+		// the match. This condition would be virtually impossible, because it
+		// would mean that the client and server were at least two steps out of
+		// sync.
+		match.failErr = fmt.Errorf("status conflict (%s -> %s) has no handler. %s",
+			match.MetaData.Status, srvStatus, logID)
+		log.Error(match.failErr)
+		match.MetaData.Proof.SelfRevoked = true
+		err := c.db.UpdateMatch(&match.MetaMatch)
+		if err != nil {
+			log.Errorf("error updating database after self revocation for no conflict handler for %s: %v", logID, err)
+		}
+	}
+
+	err := c.db.UpdateMatch(&match.MetaMatch)
+	if err != nil {
+		log.Errorf("error updating database after successful match resolution for %s: %v", logID, err)
+	}
+}
+
+// resolveMissedMakerAudit is a matchConflictResolver to handle the case when
+// our status is NewlyMatched, but the server is at MakerSwapCast.
```

Comment:
> This is only possible as the taker

That assumes that the server will never mess up. `resolveMissedMakerAudit` handles the case when we're the maker by setting `SelfRevoked` and logging an error.

> don't we need to store that?

See [the docs on `matchConflictResolver`](https://github.com/decred/dcrdex/blob/5cd4dcc08c61684082cdccf0c44ea7c40f3f8426/client/core/status.go#L93-L95).

Reply:
I wasn't referring to the actual storing *in the DB*, just updating Proof.Auth so that `resolveConflictWithServerData` can do the `UpdateMatch` with that newly-acquired data.  But I realized that there is nothing to store since we don't get the audit request to store.